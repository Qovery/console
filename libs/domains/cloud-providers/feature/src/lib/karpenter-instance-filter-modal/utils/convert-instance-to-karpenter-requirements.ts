import {
  type ClusterInstanceTypeResponseListResultsInner,
  type KarpenterNodePoolRequirement,
} from 'qovery-typescript-axios'

export function convertToKarpenterRequirements(
  instance: ClusterInstanceTypeResponseListResultsInner[]
): KarpenterNodePoolRequirement[] {
  if (!instance || instance.length === 0) {
    return []
  }

  const requirements: KarpenterNodePoolRequirement[] = []

  const uniqueValues = {
    architectures: new Set<string>(),
    families: new Set<string>(),
    sizes: new Set<string>(),
  }

  instance.forEach((instance) => {
    if (instance.architecture) {
      uniqueValues.architectures.add(instance.architecture)
    }

    if (instance.attributes?.instance_family) {
      uniqueValues.families.add(instance.attributes.instance_family)
    }
    if (instance.attributes?.instance_size) {
      uniqueValues.sizes.add(instance.attributes.instance_size)
    }
  })

  if (uniqueValues.sizes.size > 0) {
    requirements.push({
      key: 'InstanceSize',
      operator: 'In',
      values: Array.from(uniqueValues.sizes),
    })
  }

  if (uniqueValues.architectures.size > 0) {
    requirements.push({
      key: 'Arch',
      operator: 'In',
      values: Array.from(uniqueValues.architectures),
    })
  }

  if (uniqueValues.families.size > 0) {
    requirements.push({
      key: 'InstanceFamily',
      operator: 'In',
      values: Array.from(uniqueValues.families),
    })
  }

  return requirements
}
