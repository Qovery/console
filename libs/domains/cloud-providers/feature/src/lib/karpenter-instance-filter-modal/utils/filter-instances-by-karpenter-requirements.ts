import { type ClusterInstanceTypeResponseListResultsInner } from 'qovery-typescript-axios'
import { type KarpenterData } from '@qovery/shared/interfaces'

export function filterInstancesByKarpenterRequirements(
  allInstances: ClusterInstanceTypeResponseListResultsInner[],
  karpenterData: Omit<KarpenterData, 'disk_size_in_gib' | 'enabled' | 'spot_enabled'>
): ClusterInstanceTypeResponseListResultsInner[] {
  const karpenterRequirements = karpenterData.qovery_node_pools?.requirements || []

  return allInstances.filter((instance) => {
    const instanceSize = instance.attributes?.instance_size || ''
    const InstanceFamily = instance.attributes?.instance_family || ''
    const instanceArchitecture = instance.architecture || ''

    return karpenterRequirements.every((requirement) => {
      if (requirement.key === 'InstanceSize') {
        return requirement.values.includes(instanceSize)
      }

      if (requirement.key === 'Arch') {
        // Fix inconsistency between uppercase and lowercase with architecture during the migration
        return requirement.values.map((v) => v.toUpperCase()).includes(instanceArchitecture)
      }

      if (requirement.key === 'InstanceFamily') {
        return requirement.values.includes(InstanceFamily)
      }

      return false
    })
  })
}

export default filterInstancesByKarpenterRequirements
