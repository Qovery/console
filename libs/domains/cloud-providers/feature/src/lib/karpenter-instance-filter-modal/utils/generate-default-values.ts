import { type ClusterInstanceTypeResponseListResultsInner } from 'qovery-typescript-axios'
import { type KarpenterInstanceFormProps } from '../karpenter-instance-filter-modal'

export function generateDefaultValues(
  instances: ClusterInstanceTypeResponseListResultsInner[]
): Omit<KarpenterInstanceFormProps, 'default_service_architecture'> {
  // Get unique architectures
  const architectures = [...new Set(instances.map((i) => i.architecture))]

  // Get unique sizes without sorting
  const sizes = [...new Set(instances.map((i) => i.attributes?.instance_size ?? ''))].filter((i) => i !== '')

  // Group instance families by raw instance_category
  const categories: Record<string, string[]> = {}
  instances.forEach((instance) => {
    const family = instance.attributes?.instance_family || ''
    const category = instance.attributes?.instance_category || ''

    if (!categories[category]) {
      categories[category] = []
    }

    if (!categories[category].includes(family)) {
      categories[category].push(family)
    }
  })

  return {
    ARM64: architectures.includes('ARM64'),
    AMD64: architectures.includes('AMD64'),
    categories,
    sizes,
  }
}

export default generateDefaultValues
