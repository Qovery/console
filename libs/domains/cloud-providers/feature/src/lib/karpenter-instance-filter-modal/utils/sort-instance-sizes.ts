type BasicSize = 'small' | 'medium' | 'large' | 'xlarge'
type SizePriority = Record<BasicSize, number>

// Define type for the sort key tuple
type SortKey = [priority: number, multiplier: number]

export function sortInstanceSizes(sizes: string[]): string[] {
  const sizePriority: SizePriority = {
    small: 1,
    medium: 2,
    large: 3,
    xlarge: 4,
  }

  function getSortKey(size: string): SortKey {
    if (isBasicSize(size)) {
      return [sizePriority[size], 0]
    }

    // Handle xlarge variants
    if (size.includes('xlarge')) {
      // Extract number before 'xlarge'
      const number = size.replace('xlarge', '')
      // Convert empty string to 1 (for just 'xlarge')
      const multiplier = number ? parseInt(number) : 1
      return [5, multiplier]
    }

    return [0, 0]
  }

  function isBasicSize(size: string): size is BasicSize {
    return size in sizePriority
  }

  return sizes.sort((a, b) => {
    const [priorityA, multiplierA] = getSortKey(a)
    const [priorityB, multiplierB] = getSortKey(b)

    if (priorityA !== priorityB) {
      return priorityA - priorityB
    }
    return multiplierA - multiplierB
  })
}

export default sortInstanceSizes
