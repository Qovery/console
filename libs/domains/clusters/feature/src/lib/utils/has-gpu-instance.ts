import { type Cluster } from 'qovery-typescript-axios'

export const hasGpuInstance = (cluster?: Cluster) => {
  const clusterFeatureKarpenter = cluster?.features?.find((feature) => feature.id === 'KARPENTER')
  if (!clusterFeatureKarpenter) return false
  return Boolean(
    typeof clusterFeatureKarpenter?.value_object?.value === 'object' &&
      'qovery_node_pools' in clusterFeatureKarpenter.value_object.value &&
      clusterFeatureKarpenter.value_object.value.qovery_node_pools.gpu_override
  )
}
