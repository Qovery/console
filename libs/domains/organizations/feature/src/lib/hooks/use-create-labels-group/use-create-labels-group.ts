import { useMutation, useQueryClient } from '@tanstack/react-query'
import { mutations } from '@qovery/domains/organizations/data-access'
import { queries } from '@qovery/state/util-queries'

export function useCreateLabelsGroup() {
  const queryClient = useQueryClient()

  return useMutation(mutations.createLabelsGroup, {
    onSuccess(_, { organizationId }) {
      queryClient.invalidateQueries({
        queryKey: queries.organizations.labelsGroups({ organizationId }).queryKey,
      })
    },
    meta: {
      notifyOnSuccess: {
        title: 'Your labels group has been created',
      },
      notifyOnError: true,
    },
  })
}

export default useCreateLabelsGroup
