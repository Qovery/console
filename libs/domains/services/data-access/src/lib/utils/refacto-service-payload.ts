import {
  type ApplicationEditRequest,
  type ApplicationGitRepositoryRequest,
  DatabaseModeEnum,
  type HelmRequest,
  type HelmResponseAllOfSourceOneOf,
  type HelmResponseAllOfSourceOneOf1,
  type ServiceStorageStorageInner,
} from 'qovery-typescript-axios'
import { match } from 'ts-pattern'
import { isJobGitSource } from '@qovery/shared/enums'
import { type Application, type Container, type Database, type Helm, type Job } from '../domains-services-data-access'

/* 
TODO: all this following functions should be removed after the API refactoring and we need to clean it after the Services migration to React Query 
https://www.notion.so/qovery/API-improvements-b54ba305c2ee4e549eb002278c532c7f?pvs=4#7d71ea23b5fa44ca80c15a0c32ebd8da
*/

export function refactoApplication(application: Application) {
  // refacto because we can't send all git data
  if (application.git_repository) {
    application.git_repository = {
      url: application.git_repository.url,
      branch: application.git_repository.branch,
      root_path: application.git_repository.root_path,
      git_token_id: application.git_repository.git_token_id,
    }
  }

  // refacto to remove the id by storage
  if (application.storage) {
    application.storage =
      application.storage.length > 0
        ? application.storage.map((storage: ServiceStorageStorageInner) => ({
            mount_point: storage.mount_point,
            size: storage.size,
            type: storage.type,
            id: storage.id,
          }))
        : []
  }

  const applicationRequestPayload: ApplicationEditRequest = {
    name: application.name,
    storage: application.storage,
    cpu: application.cpu,
    git_repository: application.git_repository as ApplicationGitRepositoryRequest,
    build_mode: application.build_mode,
    description: application.description || undefined,
    memory: application.memory,
    auto_preview: application.auto_preview,
    auto_deploy: application.auto_deploy,
    ports: application.ports,
    dockerfile_path: application.dockerfile_path || undefined,
    healthchecks: application.healthchecks ?? {},
    buildpack_language: application.buildpack_language,
    max_running_instances: application.max_running_instances,
    min_running_instances: application.min_running_instances,
    entrypoint: application.entrypoint,
    arguments: application.arguments,
  }

  return applicationRequestPayload
}

export function refactoContainer(container: Container) {
  const containerRequestPayload = {
    name: container.name || '',
    description: container.description || '',
    storage: container.storage,
    ports: container.ports,
    cpu: container.cpu,
    memory: container.memory,
    max_running_instances: container.max_running_instances,
    min_running_instances: container.min_running_instances,
    registry_id: container.registry?.id || '',
    image_name: container.image_name || '',
    tag: container.tag || '',
    arguments: container.arguments,
    entrypoint: container.entrypoint,
    auto_preview: container.auto_preview,
    auto_deploy: container.auto_deploy,
    healthchecks: container.healthchecks,
  }

  return containerRequestPayload
}

export function refactoJob(job: Job) {
  const jobRequest = {
    name: job.name,
    description: job.description,
    cpu: job.cpu,
    memory: job.memory,
    auto_preview: job.auto_preview,
    auto_deploy: job.auto_deploy,
    max_duration_seconds: job.max_duration_seconds,
    port: job.port,
    max_nb_restart: job.max_nb_restart,
    schedule: job.schedule,
    healthchecks: job.healthchecks ?? {},
    source: {},
  }

  if (isJobGitSource(job.source)) {
    jobRequest.source = {
      docker: {
        dockerfile_path: job.source.docker?.dockerfile_path,
        git_repository: {
          url: job.source.docker?.git_repository?.url || '',
          branch: job.source.docker?.git_repository?.branch,
          root_path: job.source.docker?.git_repository?.root_path,
          git_token_id: job.source.docker?.git_repository?.git_token_id,
        },
      },
    }
  } else {
    jobRequest.source = {
      image: {
        registry_id: job.source?.image?.registry_id,
        image_name: job.source?.image?.image_name,
        tag: job.source?.image?.tag,
      },
    }
  }

  return jobRequest
}

export function refactoDatabase(database: Database) {
  const databaseRequestPayload = {
    name: database.name,
    description: database.description,
    version: database.version,
    accessibility: database.accessibility,
    cpu: database.cpu,
    memory: database.memory,
    storage: database.storage,
    instance_type: database.instance_type,
  }

  if (database.mode === DatabaseModeEnum.MANAGED) databaseRequestPayload.version = database.version

  return databaseRequestPayload
}

export function refactoHelm(helm: Helm) {
  const sourceProvider = (helm?.source as HelmResponseAllOfSourceOneOf).git ? 'GIT' : 'HELM_REPOSITORY'

  const source = match(sourceProvider)
    .with('GIT', () => {
      const gitRepository = (helm?.source as HelmResponseAllOfSourceOneOf).git?.git_repository
      return {
        git_repository: {
          url: gitRepository?.url ?? '',
          branch: gitRepository?.branch,
          root_path: gitRepository?.root_path,
          git_token_id: gitRepository?.git_token_id,
        },
      }
    })
    .with('HELM_REPOSITORY', () => {
      const helmRepository = (helm?.source as HelmResponseAllOfSourceOneOf1).repository
      return {
        helm_repository: {
          repository: helmRepository?.repository?.id,
          chart_name: helmRepository?.chart_name,
          chart_version: helmRepository?.chart_version,
        },
      }
    })
    .exhaustive()

  const helmRequestPayload: HelmRequest = {
    name: helm.name,
    description: helm.description,
    auto_preview: helm.auto_preview,
    auto_deploy: helm.auto_deploy,
    ports: helm.ports,
    allow_cluster_wide_resources: helm.allow_cluster_wide_resources,
    arguments: helm.arguments,
    values_override: helm.values_override,
    source,
  }

  return helmRequestPayload
}
