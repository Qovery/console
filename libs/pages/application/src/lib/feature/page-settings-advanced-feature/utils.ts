import { ApplicationAdvancedSettings } from 'qovery-typescript-axios'
import { ApplicationEntity } from '@console/shared/interfaces'

export function initFormValues(keys: string[], application: ApplicationEntity): { [key: string]: string } {
  const values: { [key: string]: string } = {}
  keys.forEach((key) => {
    if (application.advanced_settings?.current_settings) {
      values[key] =
        application.advanced_settings.current_settings[key as keyof ApplicationAdvancedSettings]?.toString() || ''
    }
  })

  return values
}
