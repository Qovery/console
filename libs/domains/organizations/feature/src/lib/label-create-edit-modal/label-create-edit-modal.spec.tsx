import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
import { renderWithProviders, screen } from '@qovery/shared/util-tests'
import * as useCreateLabelsGroup from '../hooks/use-create-labels-group/use-create-labels-group'
import * as useEditLabelsGroup from '../hooks/use-edit-labels-group/use-edit-labels-group'
import LabelCreateEditModal, { type LabelCreateEditModalProps } from './label-create-edit-modal'

const useCreateLabelsGroupMockSpy = jest.spyOn(useCreateLabelsGroup, 'useCreateLabelsGroup') as jest.Mock
const useEditLabelsGroupMockSpy = jest.spyOn(useEditLabelsGroup, 'useEditLabelsGroup') as jest.Mock

const props: LabelCreateEditModalProps = {
  organizationId: '0000-0000-0000',
  onClose: jest.fn(),
}

describe('LabelCreateEditModal', () => {
  beforeEach(() => {
    useCreateLabelsGroupMockSpy.mockReturnValue({
      mutateAsync: jest.fn().mockResolvedValue({ id: '000' }),
    })
    useEditLabelsGroupMockSpy.mockReturnValue({
      mutateAsync: jest.fn(),
    })
  })

  it('should match with snatpshot', () => {
    const { baseElement } = renderWithProviders(wrapWithReactHookForm(<LabelCreateEditModal {...props} />))
    expect(baseElement).toMatchSnapshot()
  })

  it('should submit the form to create a label group', async () => {
    const { userEvent } = renderWithProviders(wrapWithReactHookForm(<LabelCreateEditModal {...props} />))

    const inputName = screen.getByLabelText('Group name')
    await userEvent.type(inputName, 'my-name')

    const inputKey = screen.getByTestId('labels.0.key')
    await userEvent.type(inputKey, 'key')

    const inputValue = screen.getByTestId('labels.0.value')
    await userEvent.type(inputValue, 'value')

    const btn = screen.getByRole('button', { name: 'Create' })
    expect(btn).toBeEnabled()

    await userEvent.click(btn)

    expect(useCreateLabelsGroupMockSpy().mutateAsync).toHaveBeenCalledWith({
      organizationId: '0000-0000-0000',
      labelsGroupRequest: {
        name: 'my-name',
        labels: [
          {
            key: 'key',
            value: 'value',
            propagate_to_cloud_provider: false,
          },
        ],
      },
    })

    expect(props.onClose).toHaveBeenCalled()
  })

  it('should submit the form to edit a labels group', async () => {
    const { userEvent } = renderWithProviders(
      wrapWithReactHookForm(
        <LabelCreateEditModal
          {...props}
          isEdit
          labelsGroup={{
            id: '1111-1111-1111',
            created_at: '',
            updated_at: '',
            name: 'name',
            labels: [
              {
                key: 'key',
                value: 'value',
                propagate_to_cloud_provider: false,
              },
            ],
          }}
        />
      )
    )

    const inputName = screen.getByLabelText('Group name')
    await userEvent.clear(inputName)
    await userEvent.type(inputName, 'my-name')

    const btn = screen.getByRole('button', { name: 'Confirm' })
    expect(btn).toBeEnabled()

    await userEvent.click(btn)

    expect(useEditLabelsGroupMockSpy().mutateAsync).toHaveBeenCalledWith({
      organizationId: '0000-0000-0000',
      labelsGroupId: '1111-1111-1111',
      labelsGroupRequest: {
        name: 'my-name',
        labels: [
          {
            key: 'key',
            value: 'value',
            propagate_to_cloud_provider: false,
          },
        ],
      },
    })

    expect(props.onClose).toHaveBeenCalled()
  })
})
