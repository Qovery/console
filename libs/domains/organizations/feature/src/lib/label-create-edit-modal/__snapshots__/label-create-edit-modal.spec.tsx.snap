// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelCreateEditModal should match with snatpshot 1`] = `
<body>
  <div>
    <div
      class="p-5"
    >
      <h2
        class="h4 max-w-sm truncate text-neutral-400 dark:text-neutral-50"
      >
        Create label group
      </h2>
      <p
        class="mt-2 text-sm text-neutral-350 dark:text-neutral-50"
      >
        You will have the possibility to modify the parameters once created.
      </p>
      <form
        class="mt-6"
      >
        <div
          class="mb-5"
          data-testid="input-text-wrapper"
        >
          <div
            class="relative"
          >
            <div
              aria-label="input-container"
              class="input     "
            >
              <div
                class=""
              >
                <label
                  class="translate-y-2 text-sm"
                  for="Group name"
                >
                  Group name
                </label>
                <input
                  class="input__value "
                  data-testid="input-text"
                  id="Group name"
                  name="name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <span
          class="mb-2 block text-sm text-neutral-400"
        >
          Add label key value
        </span>
        <div
          class="rounded border border-neutral-250 bg-neutral-100 px-4 py-3"
        >
          <ul>
            <li
              class="mb-3 grid grid-cols-[6fr_6fr_1fr_1fr] gap-x-2"
            >
              <span
                class="text-sm font-medium text-neutral-350"
              >
                Label keys
              </span>
              <span
                class="pl-10 text-sm font-medium text-neutral-350"
              >
                Value
              </span>
              <span
                class="relative left-7 flex items-center gap-1 whitespace-nowrap text-sm font-medium text-neutral-350"
              >
                Propagate as tag
                <span
                  class=""
                  data-state="closed"
                >
                  <i
                    aria-hidden="true"
                    class="fa-solid fa-circle-info "
                  />
                </span>
              </span>
              <span />
            </li>
            <li
              class="mb-3 last:mb-0"
            >
              <div
                class="grid grid-cols-[6fr_6fr_1fr_1fr] items-center gap-x-2"
              >
                <div
                  class=""
                  data-testid="input-small-wrapper"
                >
                  <div
                    class="input input--small flex-grow"
                    data-testid="input"
                  >
                    <label
                      class="hidden"
                    />
                    <input
                      class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                      data-testid="labels.0.key"
                      name="labels.0.key"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class=""
                  data-testid="input-small-wrapper"
                >
                  <div
                    class="input input--small flex-grow"
                    data-testid="input"
                  >
                    <label
                      class="hidden"
                    />
                    <input
                      class="absolute left-0 top-0 h-full w-full rounded px-2 text-sm text-neutral-400 placeholder:text-neutral-350"
                      data-testid="labels.0.value"
                      name="labels.0.value"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="flex text-sm  items-start  "
                  data-testid="input-toggle"
                >
                  <div
                    aria-label="toggle-btn"
                    class="inline-flex items-center justify-between cursor-pointer"
                    data-testid="input-toggle-button"
                  >
                    <input
                      class="hidden"
                      type="checkbox"
                      value="false"
                    />
                    <div
                      aria-label="bg"
                      class="w-8 h-4.5 flex items-center rounded-full p-0.5  bg-neutral-300"
                    >
                      <div
                        aria-label="circle"
                        class="w-3.5 h-3.5 transform rounded-full bg-white shadow-lg  "
                      />
                    </div>
                  </div>
                </div>
                <button
                  class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-9 px-3 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    class="fa-solid fa-trash text-sm text-neutral-400"
                  />
                </button>
              </div>
            </li>
          </ul>
          <button
            class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-xs h-7 px-2 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100 mt-3"
            type="button"
          >
            Add label
            <i
              aria-hidden="true"
              class="fa-solid fa-plus ml-2"
            />
          </button>
        </div>
        <div
          class="mt-6 flex justify-end gap-3"
        >
          <button
            class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 hover:[&:not(:focus-visible):not(:disabled)]:shadow-sm active:[&:not(:focus-visible)]:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-transparent text-neutral-350 hover:[&:not(:active):not(:focus-visible)]:border-neutral-300 hover:[&:not(:disabled)]:text-neutral-400 dark:hover:[&:not(:disabled)]:text-neutral-250 focus-visible:text-neutral-400 active:bg-neutral-150 dark:active:bg-transparent active:text-neutral-400 disabled:text-neutral-300 disabled:bg-neutral-150 disabled:border-none"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
          <button
            class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-11 px-5 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
            data-testid="submit-button"
            disabled=""
            type="submit"
          >
            Create
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`;
